### ISSUE TITLE
Hide Passwords

### ISSUE THREAD

**Issue by amouat** [xolox/python-qpass#1]
This project is almost exactly what I'm looking for. I literally searched for "pass fuzzy matching". qpass worked straight away, the only problem is that it shows passwords in clear text on the terminal - it would be great to add an option to hide them and just copy to the clipboard.

**Comment by xolox** [comment]
Hi Adrian and thanks for the feedback, it's nice to hear that someone finds this useful! :slightly_smiling_face:

Passwords should only be shown on the terminal when qpass detects that the clipboard is not supported, reasoning that without clipboard support the only way to share the password is by showing it in clear text on the terminal. But of course the clipboard is the preferred method (an understatement :smile:).

So it sounds to me like this detection doesn't work correctly on your system. Here's the relevant code:

```python
def is_clipboard_supported():
    """
    Check whether the clipboard is supported.

    :returns: :data:`True` if the clipboard is supported, :data:`False` otherwise.
    """
    return platform.system().lower() == 'darwin' or bool(os.environ.get('DISPLAY'))
```

This code assumes that the clipboard is always supported on macOS and supported on other UNIX systems when the environment variable `$DISPLAY` is set.

Could you help me figure out why the clipboard support detection doesn't work on your system?

**Comment by amouat** [comment]
Hmm, I did wonder why you chose to display passwords :)

The clipboard is working; I can paste passwords successfully (otherwise I'd be completely stuck as I'm not showing the password). I'm running Ubuntu. 

I just had a look at the code. The problem is that it outputs the rest of the text after the first line, and my pass output look like this:

    secret_password
    ---
    site_name
    username: myuser
    password: secret_password
    url: https:/bla

Note the second password field. This might be a hangover from importing data to pass from another password manager. So, I'd either need qpass to strip any line starting "password:" or not to output details by default.

I'm not sure how common it is for users to have "password:" lines, so I understand if you don't want to add extra code just for me. I might update my own fork to instead strip password fields, so I can still get the usernames easily without displaying passwords.

**Comment by xolox** [comment]
Hi Adrian and thanks for following up with an explanation!

While there isn't a lot of structure to the format of password store entries in `pass`, having the password on the first line definitely is part of the structure :slightly_smiling_face:. While writing `qpass` I guess I simply assumed that users wouldn't repeat the password after the first line.

Anyway, "hangover from import" makes sense and it does highlight the missing features you pointed out:

1. **Hiding the rest of the text (after the first line):** In https://github.com/xolox/python-qpass/commit/7ae7e018f7f82ffd6ca71e6f9ff953413cd75c63 I changed the command line interface so that the rest of the text is not shown when you use the `-q`, `--quiet` option. This is actually useful for me because I have password store entries that contain 2FA recovery codes in the rest of the text and there's usually no point in printing these to the terminal.

2. **Hiding specific lines in the rest of the text:** In https://github.com/xolox/python-qpass/commit/5a80dc79429d63cebb35f76f5807706261b984c2 I added the `-f`, `--filter` option that takes regular expressions and hides lines matching the given regular expression(s). This should make it possible to hide specific lines like the `password: ...` lines in your password store entries. A command line like `qpass -f '^password:' ...` should do the trick.

Currently `qpass` doesn't have support for configuration files and of course you don't want to have to type out the  `-f` option every time, but given that `qpass` is a command line program I guess a simple shell alias could make this easier to use? I personally use a shell alias to provide multiple `-p` options to `qpass` so that I can query all of my password stores at the same time.

These new features are part of `qpass` 2.1 which I just released to PyPI and GitHub. I hope this helps!
